components/camel-mongodb/src/main/java/org/apache/camel/component/mongodb/verifier/MongoComponentVerifierExtension.java

```diff
@@ -0,0 +1,101 @@
+import org.apache.camel.component.extension.verifier.ResultErrorHelper;
+import org.apache.camel.component.mongodb.conf.MongoConfiguration;
+import org.bson.Document;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static org.apache.camel.util.CastUtils.cast;
+
+public class MongoComponentVerifierExtension extends DefaultComponentVerifierExtension {
+    private static final Logger LOG = LoggerFactory.getLogger(MongoComponentVerifierExtension.class);
+
+    private static final int CONNECTION_TIMEOUT = 2000;
+
+    public MongoComponentVerifierExtension() {
+        super("mongodb");
+    }
+
+    @Override
+    public Result verifyParameters(Map<String, Object> parameters) {
+        ResultBuilder builder = ResultBuilder.withStatusAndScope(Result.Status.OK, Scope.PARAMETERS)
+                .error(ResultErrorHelper.requiresOption("host", parameters))
+                .error(ResultErrorHelper.requiresOption("user", parameters))
+                .error(ResultErrorHelper.requiresOption("password", parameters));
+        return builder.build();
+    }
+
+    @Override
+    public Result verifyConnectivity(Map<String, Object> parameters) {
+        return ResultBuilder.withStatusAndScope(Result.Status.OK, Scope.CONNECTIVITY)
+                .error(parameters, this::verifyCredentials)
+                .build();
+    }
+
+    private void verifyCredentials(ResultBuilder builder, Map<String, Object> parameters) {
+        MongoConfiguration mongoConf = new MongoConfiguration(cast(parameters));
+        MongoClientOptions.Builder optionsBuilder = MongoClientOptions.builder();
+        // Avoid retry and long timeout
+        optionsBuilder.connectTimeout(CONNECTION_TIMEOUT);
+        optionsBuilder.serverSelectionTimeout(CONNECTION_TIMEOUT);
+        optionsBuilder.maxWaitTime(CONNECTION_TIMEOUT);
+        MongoClientURI connectionURI = new MongoClientURI(mongoConf.getMongoClientURI(), optionsBuilder);
+
+        LOG.info("Testing connection against {}", connectionURI);
+        try (MongoClient mongoClient = new MongoClient(connectionURI)) {
+            // Just ping the server
+            mongoClient.getDatabase(connectionURI.getDatabase()).runCommand(Document.parse("{ ping: 1 }"));
+            LOG.info("Testing connection successful!");
+        } catch (MongoSecurityException e) {
+            ResultErrorBuilder errorBuilder = ResultErrorBuilder.withCodeAndDescription(
+                    VerificationError.StandardCode.AUTHENTICATION,
+                    String.format("Unable to authenticate %s against %s authentication database!", mongoConf.getUser(), mongoConf.getAdminDB()));
+            builder.error(errorBuilder.build());
+        } catch (MongoTimeoutException e) {
+            // When there is any connection exception, the driver tries to reconnect until timeout is reached
+            // wrapping the original socket exception into a timeout exception
+            String description = null;
```

Comment:
also I don't think `null` is needed here

Reply:
Yes, `else` branch is taking care to initialize if not done before.